### DISCUSSION TITLE
Build Official Documentation Website

### DISCUSSION THREAD

**Discussion by icedrone** [OrcaSlicer/OrcaSlicer#7440] [Ideas]
I propose creating an official documentation website using an open-source tool that generates docs from Markdown, similar to a wiki. This approach has several advantages over a standard wiki setup:

1. **Full-Text Search and Flexibility** - Provides better search capabilities and flexibility in organizing content.
2. **Improved Search Indexing** - An official site would likely rank better on Google, potentially pushing down third-party sites like "orcaslicer.net" in search results.
3. **Enhanced Content Control** - As an official resource, we would have more control over content, allowing us to report misleading information that appears elsewhere more effectively.

For this project, I suggest opening a new repository called `OrcaSlicer-docs`. Alternatively, we could adopt a monorepo approach if preferred.

For hosting, we can use [Vercel](https://vercel.com), which offers free options for hobby projects. For the domain, we could use a free Vercel subdomain or, if OrcaSlicer already has an existing domain, that would be even better.

As for the documentation engine, I recommend [FumaDocs](https://github.com/fuma-nama/fumadocs), an open-source project we use at my company for [Myrror Security docs](https://docs.myrror.security/docs). It includes advanced search functionality powered by Algolia, which has worked exceptionally well for us.

I’d be more than happy to take on this project if it’s something the community is interested in and willing to support.

**Comment by SoftFever** [comment]
Thank you for your suggestions. 
I plan to transition to a more traditional WIKI website to host the Orca Slicer wiki and documentation. 
This change will facilitate easier collaboration and contribution from the community compared to the current GitHub WIKI, which operates similarly to your proposal. 

I am in the process of setting up a new wiki on Miraheze (currently available at [https://orcaslicer.miraheze.org/wiki/Main_Page ](https://orcaslicer.miraheze.org/wiki/Main_Page)and soon to be accessible at [wiki.orcaslicer.com ](http://wiki.orcaslicer.com/)). 
The wiki migration has taken a bit longer than I anticipated due to some personal matters. However, it should be ready in time for the release of OrcaSlicer version 2.3.

**Reply by icedrone** [reply]
Hi @SoftFever ,

First of all, I’m really excited to hear about this progress and the new direction you're considering.

I’m not very familiar with the miraheze.org project, but based on what I’ve seen from your link and other sources online, it feels a bit outdated. Modern alternatives, especially those based on markdown, offer better full-text search, a more user-friendly interface, and can significantly improve collaboration and contributions to the main project. Keeping everything centralized could help contributors stay within the ecosystem they’re already comfortable with, instead of directing them to an unfamiliar external platform.

As you’re probably aware, a well-designed documentation website can elevate the project to new heights by engaging both new and existing users. I’ve also explored other projects that prioritize good UX/UI in their documentation, and the difference is remarkable. Moving to a traditional wiki-like experience could potentially replicate the challenges of GitHub's current wiki, which from personal experience can be a bit overwhelming and unintuitive for beginners trying to find their way.

Additionally, improving the documentation could serve as a great entry point for newcomers to make their first contribution to the project. If we shift to an external wiki, we might lose that advantage by fragmenting the community and moving them away from GitHub—which I personally don’t think would be the best approach.

That said, if the decision is to move forward with a wiki-based experience, despite my arguments, I’d be happy to help in any way I can to support the project!

**Comment by rastacalavera** [comment]
Hello!

I've been using Orca Slicer for several months now and have found the Wiki helpful but think it could benefit from some contributions. I've been looking through the repo to try to find out how to properly contribute to it but came up empty handed. Sorry to resurrect an old thread but this seemed like the logical place to comment. I visited the sites above and they are deactivated due to lack of activity. 

Is there a way I can put a pr in for the GitHub wiki? If not, how can I contribute to the one that is under construction at the other pages?

**Reply by ianalexis** [reply]
I've been working on updating the wiki all this time and created a guide on how to do it.
https://github.com/SoftFever/OrcaSlicer/wiki/How-to-wiki

If you look at the PR list on the wiki, you can see how I've been working and how many users participated in the process and were co-authors.
